src/app/OperationalDeviceProxy.cpp

```diff
@@ -234,13 +275,20 @@ void OperationalDeviceProxy::HandleCASEConnectionFailure(void * context, CASECli
                    ChipLogError(Controller, "HandleCASEConnectionFailure was called while the device was not initialized"));
     VerifyOrReturn(client == device->mCASEClient, ChipLogError(Controller, "HandleCASEConnectionFailure for unknown CASEClient"));
 
+    //
+    // We don't need to reset the state all the way back to NeedsAddress since all that transpired
+    // was just CASE connection failure. So let's re-use the cached address to re-do CASE again
+    // if need-be.
+    //
     device->MoveToState(State::Initialized);
 
     device->CloseCASESession();
-    device->DequeueConnectionSuccessCallbacks(/* executeCallback */ false);
-    device->DequeueConnectionFailureCallbacks(error, /* executeCallback */ true);
-    // Do not touch device anymore; it might have been destroyed by a failure
+    device->DequeueConnectionCallbacks(error);
+
+    //
+    // Do not touch this instance anymore; it might have been destroyed by a failure
```

Comment:
```suggestion
    // Do not touch device anymore; it might have been destroyed by a failure
```
This is a static class method, so there is no "this instance".  `device` is the thing that might be dead.